docs/decisions/0003-add-access-policy-functionality.rst

```diff
@@ -0,0 +1,313 @@
+See the `Enterprise Subsidy Initial ADR`_ for more context.
+
+The business decisions around which learners are allowed to redeem value for content can be varied and
+sometimes complex, depending on the nature of our contract with an enterprise, for example:
+
+- Enterprise Learners should only be able to spend 500 dollars on a particular type of content during the life of their subsidy.
+- Enterprise Learners should only be able to enroll in courses that are in a catalog associated with an enterprise.
+- Enterprise Learners should only be able to enroll in 3 courses, maximum, during the lifetime of their subsidy.
+
+We currently do not have an application that allows us to represent and enforce these rules - they are instead enforced
+across frontend and backend code - and therefor cannot ascertain how such rules are applied without examining
+multiple codebases. Keep in mind that multiple of these allowable business decisions, or "policies", may be applicable at a given
+time for a given enterprise.
+
+In writing an access policy application, we'll be able to command and query who is allowed to redeem
+subsidy value, from which allowed set of content, via what access method.  It should also support our
+general compliance requirements, protecting our business reputation and general business account veractiy.
+
+Decision
+********
+
+We will build a Subsidy Access Policy application to that meets the qualities described above.  It will
+be a new Django application in the ``enterprise-access`` service.  It will broadly include:
+
+- Models that persist data
+- A Python API that defines business logic, through which consumers interact with the models.
+- A REST API defining views that interact with the Python API (aka a “web service”)
+- A producer/emitter of asynchronous events sent to some message broker.
+
+The base access policy model should be composed of references to an enterprise subsidy, an enterprise catalog,
+an access method (e.g. direct learner enrollment, or Browse & Request), and optionally, the total value allowed
+to be redeemed via the policy (that is, the maximum number of dollars or seats allowed to be consumed).  In the future,
+we suspect that it will also be composed of a reference to an enterprise group - a component that does not
+yet exist in the enterprise system.
+
+Types of policies
+=================
+
+Modeling policies via Single Table Inheritance
+----------------------------------------------
+
+The base ``SubsidyAccessPolicy``  model defines the most core business logic (i.e. method implementations)
+for all types of access policies.  It implements ``can_redeem()``, ``redeem()``, and ``has_redeemed()`` methods,
+which take into account the inclusion of some content key in the policy’s catalog; whether there is
+enough remaining value in the subsidy; and whether the given user belongs to the policy’s group
```

Comment:
Should we mention the other base check that the subsidy itself is active or expired?